tidb-server/main.go

```diff
@@ -548,4 +548,5 @@ func cleanup() {
 		svr.KillAllConnections()
 	}
 	closeDomainAndStorage()
+	svr.CleanupSocketFile()
```

Comment:
This may be out of the scope of this PR, but it seems that at least for this cleanup it should happen in the context of a defer that recovers from a panic.

Reply:
It now handles correctly in server `Close()`